src/main/java/eu/mihosoft/vrl/v3d/CSG.java

```diff
@@ -1100,13 +1337,16 @@ public MeshContainer toJavaFXMeshSimple() {
 
     /**
      * Returns the bounds of this csg.
+     * SIDE EFFECT bounds is created and simply returned if existing

```

Comment:
Your approach does not always return correct bounds (Polygons and Vertices are mutable).